File: app/src/main/java/joy/liu/com/hencoderPractice/practice7/Sample01ArgbEvaluatorLayout.java
package joy.liu.com.hencoderPractice.practice7;

import android.animation.ArgbEvaluator;
import android.animation.ObjectAnimator;
import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.widget.Button;
import android.widget.RelativeLayout;

import joy.liu.com.practicedraw1.R;

/**
 * Created by joy on 2017/11/21.
 */

public class Sample01ArgbEvaluatorLayout extends RelativeLayout {
    Sample01ArgbEvaluatorView view;
    Button animateBt;
    public Sample01ArgbEvaluatorLayout(Context context) {
        super(context);
    }

    public Sample01ArgbEvaluatorLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public Sample01ArgbEvaluatorLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onAttachedToWindow() {
        super.onAttachedToWindow();
        view = findViewById(R.id.objectAnimatorView);
        animateBt = findViewById(R.id.animateBt);

        animateBt.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                ObjectAnimator animator = ObjectAnimator.ofInt(view, "color", 0xffff0000, 0xff00ff00);
                animator.setEvaluator(new ArgbEvaluator());
                animator.setDuration(2000);
                animator.start();
            }
        });
    }
}

File: app/src/main/java/joy/liu/com/hencoderPractice/practice7/Sample01ArgbEvaluatorLayout_.kt
package joy.liu.com.hencoderPractice.practice7

import android.animation.ArgbEvaluator
import android.animation.ObjectAnimator
import android.content.Context
import android.util.AttributeSet
import android.widget.Button
import android.widget.RelativeLayout
import joy.liu.com.practicedraw1.R


/**
 * Created by joy on 2017/11/16.
 */
class Sample01ArgbEvaluatorLayout_ : RelativeLayout {
    var view: Sample01ArgbEvaluatorView_? = null
    var animateBt: Button? = null


    constructor(context: Context) : super(context) {
        initData()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        initData()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        initData()
    }

    private fun initData() {

    }

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        view = findViewById(R.id.objectAnimatorView)
        animateBt = findViewById(R.id.animateBt)

        animateBt?.setOnClickListener({
            val objectAnimator = ObjectAnimator.ofInt(view, "color", 0xff0000, 0x00ff00)
            objectAnimator.setEvaluator(ArgbEvaluator())
            objectAnimator.duration=2000
            objectAnimator.start()
        })
    }

}